src/lib/operators/projection.cpp

```diff
@@ -127,10 +137,11 @@ std::shared_ptr<const Table> Projection::_on_execute() {
     for (auto column_id = ColumnID{0}; column_id < expression_count; ++column_id) {
       // In this loop, we perform all projections that only forward an input column sequential.
       const auto& expression = expressions[column_id];
-      if (expression->type != ExpressionType::PQPColumn) {
+      if (expression->type != ExpressionType::PQPColumn || pqp_columns_to_evaluate[column_id]) {
```

Comment:
It feels as if the logic on which column should be forwarded is too spread out. If we can just use `contains`, we shouldn't have to check for `PQPColumn` or `output_table_type == TableType::References` all over the place.

Reply:
> If we can centralize the "which expressions are to be forwarded" logic, this might clean up the projection operator quite a bit - potentially even beyond what we currently have in the master.

Looking at the code two days later, I can only agree. Will take a look at it.